File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        int i=0;
        int n=s.length();
        bool sign=false,neg=false;
        long long res=0;
        while(i<n && s[i]==' ') i++;
        if(s[i]=='-' || s[i]=='+'){
            sign=true;
            neg=s[i]=='-'?true:false;
            i++;
        }
        for(;i<n && res<INT_MAX;i++){
            if(s[i]=='-' || s[i]=='+'){
                break; 
            }
            else if(isdigit(s[i])) res=res*10+(s[i]-'0');
            else break;
        }

        return neg?-res<INT_MIN?INT_MIN:-res:res>INT_MAX?INT_MAX:res;
    }
};

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        s=s.strip() 
        if not(s): 
            return 0
        neg = False 
        if s[0]=='-' or s[0]=='+':
            if s[0]=='-':
                neg=True 
            s=s[1:]

        res = 0 
        for val in s: 
            if val.isdigit(): 
                res= res*10+int(val)
            else: 
                break
        res =-res if neg else res
        if res<-2**31:
            return -2**31
        elif res>2**31-1:
            return 2**31-1 
        return res